game_template_1.0

```diff
@@ -0,0 +1,264 @@
+/*
+------------------------------------------------------------------
+  Title: Text Based RPG - Name pending
+  Creators: NAMES HERE
+  Description:
+    You are on a mission to escape by rescuing a copilot
+    and repair the spaceship's engines. Gameplay includes
+    enemy combat, navigate through different rooms,
+    encounter challenges, and solve puzzles to succeed.
+
+    Rooms:
+    - Medical Bay: Starting point with supplies.
+    - Sleeping Quarters: Find the copilot.
+    - Storage Room: Encounter a space pirate.
+    - Engine Room: Solve a puzzle to fix the engines.
+    - Cockpit: Check mission success conditions.
+
+    Game Mechanics:
+    - Movement: Choose options to move between rooms.
+    - Combat: Dice rolls determine outcomes against enemies.
+    - Puzzle: Solve puzzles to progress. WIP
+    - Success: Check if copilot is safe and engines are fixed.
+-------------------------------------------------------------------
+*/
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <cstdlib>
+#include <ctime>
+
+using namespace std;
+
+// Function to read text from a file
+// Takes a filename as an argument and returns the file's content as a string
+string readFromFile(const string& filename)
+{
+    ifstream file(filename);
+    if (!file)
+    {
+        cerr << "Error: Could not open file " << filename << endl;
+        return "";
+    }
+    ostringstream content;
+    content << file.rdbuf();
+    file.close();
+    return content.str();
+}
+
+// Function to wait for any key press
+// Pauses the game until the player presses a key
+void waitForKeyPress()
+{
+    cout << "Press any key to roll the dice...\n";
+    cin.ignore();
+    cin.get();
+}
+
+// Class for the Copilot
```

Comment:
I like the addition of a Copilot class! If we decide to expand functionality, this lends itself well to that. In addition, I think it would be helpful to add a Player class. Here is my proposal for this:

```cpp
#include <cstring>
#include <iostream>

using namespace std;

enum location {ENGINE_ROOM, HALLWAY, COCKPIT, STORAGE, MEDICAL_BAY, SLEEPING_QUARTERS};

class Player {
    public:
        bool get_key() {
            return key;
        }
        void set_key(bool val) {
            key = val;
        }
        bool get_enginePiece() {
            return enginePiece;
        }
        void set_enginePiece(bool val) {
            enginePiece = val;
        }
        location get_location() {
            return playerLocation;
        }
        void set_location(location val) {
            playerLocation = val;
        }
        string get_name() {
            return name;
        }
        void set_name(string val) {
            name = val;
        }
    private:
        bool key;
        bool enginePiece;
        location playerLocation;
        string name;
};
```

Depending on our needs, we could use location tracking for expansion. Each switch statement would change the location value and the enum location would need to be Global.  For the engine room item tracking and the storage room key tracking, we can use player data members rather than the main function variables.